### PR TITLE
Add option to disable dock

### LINKED ISSUE #2327: Hide Dock Option
Feature Request:
Pls give an option to disable the dock completely, so that we users can utilize that space too, adding widgets, etc...

### LINKED ISSUE #3839: [FEATURE] Option to disable dock
### Describe the feature

Option to fully disable the dock and the search widget.

### Why would it be useful to add?

This function would allow the user more space on the main screen to place widgets or app icons.

### Did the feature exist in Lawnchair v2? (Play Store version)

No

### Example(s)

Like the Neo Launcher implementation.

Screenshot_20231209-111511_Neo Launcher

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MrSluffy**
## Description

- Closes #3839 
- Closes #2327

## Type of change

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:white_check_mark: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: lawnchair/res/values/strings.xml [hunk 1]
```diff
@@ -285,6 +285,7 @@
 
     <!-- DockPreferences -->
     <!-- <string name="dock_label" /> -->
+    <string name="show_hotseat_title">Show/Hide Dock</string>
```
Comment (SuperDragonXD) [inline lawnchair/res/values/strings.xml]:
```suggestion
    <string name="show_hotseat_title">Enable Dock</string>
```

File: lawnchair/src/app/lawnchair/ui/preferences/destinations/DockPreferences.kt [hunk 2]
```diff
@@ -57,46 +57,55 @@ fun DockPreferences() {
...
-                        )
+        val isHotseatEnabled = prefs2.isHotseatEnabled.getAdapter()
+        val hotseatModeAdapter = prefs2.hotseatMode.getAdapter()
+        PreferenceGroup(heading = stringResource(id = R.string.general_label)) {
+            SwitchPreference(
+                adapter = isHotseatEnabled,
+                label = stringResource(id = R.string.show_hotseat_title),
+            )
+        }
+        if (isHotseatEnabled.state.value) {
```
Comment (SuperDragonXD) [inline lawnchair/src/app/lawnchair/ui/preferences/destinations/DockPreferences.kt]: Can we use `ExpandAndShrink` instead?

```suggestion
        ExpandAndShrink (visible = isHotseatEnabled.state.value) {
```

**Review by SuperDragonXD — COMMENTED** [review]
I'm requesting a string change as always.

Also, I'm planning to create a `MainSwitchPreference` components (similar to the  main switches found in Android Settings: i.e. Wifi, Location), but the current `SwitchPreference` should be good for now.

File: src/com/android/launcher3/DeviceProfile.java [hunk 3]
```diff
@@ -314,6 +314,8 @@ public class DeviceProfile {
 
     private final DeviceProfileOverrides.TextFactors mTextFactors;
 
+    PreferenceManager2 preferenceManager2;
```
Comment (Goooler) [inline src/com/android/launcher3/DeviceProfile.java]:
```suggestion
    private final PreferenceManager2 preferenceManager2;
```

File: src/com/android/launcher3/Workspace.java [hunk 4]
```diff
@@ -1390,6 +1390,9 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt) {
     public void showPageIndicatorAtCurrentScroll() {
         if (mPageIndicator != null) {
             mPageIndicator.setScroll(getScrollX(), computeMaxScroll());
+            var isHotseatEnabled = PreferenceExtensionsKt.firstBlocking(mPreferenceManager2.isHotseatEnabled());
+            mPageIndicator.setVisibility(isHotseatEnabled ? VISIBLE : INVISIBLE);
+            mPageIndicator.setAlpha(isHotseatEnabled ? 1 : 0);
```
Comment (suphon-t) [inline src/com/android/launcher3/Workspace.java]: No need to set alpha if it's already invisible, but why hide this too?
```suggestion
            mPageIndicator.setAlpha(isHotseatEnabled ? 1 : 0);
```

File: src/com/android/launcher3/DeviceProfile.java [hunk 5]
```diff
@@ -795,7 +797,7 @@ private void updateHotseatSizes(int hotseatIconSizePx) {
         // radius.
         hotseatCellHeightPx = getIconSizeWithOverlap(hotseatIconSizePx);
 
-        var space = Math.abs(hotseatCellHeightPx / 2);
+        var space = Math.abs(hotseatCellHeightPx / 2) - 10;
```
Comment (suphon-t) [inline src/com/android/launcher3/DeviceProfile.java:800]: Should this only apply if the dock is disabled?

File: src/com/android/launcher3/DeviceProfile.java [hunk 6]
```diff
@@ -810,6 +812,10 @@ private void updateHotseatSizes(int hotseatIconSizePx) {
                     + hotseatBarBottomSpacePx
                     + space;
         }
+       var isHotseatEnabled = PreferenceExtensionsKt.firstBlocking(preferenceManager2.isHotseatEnabled());
```
Comment (suphon-t) [inline src/com/android/launcher3/DeviceProfile.java]: Formatting
```suggestion
        var isHotseatEnabled = PreferenceExtensionsKt.firstBlocking(preferenceManager2.isHotseatEnabled());
```

File: src/com/android/launcher3/Workspace.java (hunk 4 above)
Comment (MrSluffy) [reply src/com/android/launcher3/Workspace.java]: Your right 

the pageindicator is showing at bottom, we need to hide it

File: src/com/android/launcher3/DeviceProfile.java (hunk 5 above)
Comment (MrSluffy) [reply src/com/android/launcher3/DeviceProfile.java:800]: if enable, just a little icons alignment

**Review by suphon-t — APPROVED** [review]

**Pull request merged**